File: euler/problem-2/main.py
def sum_of_even_fibo(limit):
    a, b = 1, 2
    total = 0
    while a <= limit:
        if a % 2 == 0:
            total += a
        a, b = b, a + b
    return total
        

limit = 4000000
res = sum_of_even_fibo(limit)
print(res)

File: euler/problem-2/main.ts
function sum_of_even_fibo(limit: number): number {
    let a: number = 1
    let b: number = 2
    let total: number = 0

    while (a <= limit) {
        if (Math.floor(a % 2) === 0) {
            total = total + a
        }
        let aux: number = a + b
        a = b
        b = aux
    }
    return total
}

let limit: number = 4000000
let res: number = sum_of_even_fibo(limit)
console.log(res);
